src/Analyzers/Orchestration/OrchestrationAnalyzer.cs

```diff
@@ -0,0 +1,285 @@
+
+                IEnumerable<MethodDeclarationSyntax> methodSyntaxes = methodSymbol.GetSyntaxNodes();
+                foreach (MethodDeclarationSyntax rootMethodSyntax in methodSyntaxes)
+                {
+                    FindInvokedMethods(ctx.SemanticModel, rootMethodSyntax, methodSymbol, orchestration, result);
+                }
+            },
+                SyntaxKind.ClassDeclaration);
+
+            // look for OrchestratorFunc Orchestrations
+            context.RegisterOperationAction(
+                ctx =>
+            {
+                if (ctx.Operation is not IInvocationOperation invocation)
+                {
+                    return;
+                }
+
+                if (!SymbolEqualityComparer.Default.Equals(invocation.Type, knownSymbols.DurableTaskRegistry))
+                {
+                    return;
+                }
+
+                // there are 8 AddOrchestratorFunc overloads
+                if (invocation.TargetMethod.Name != "AddOrchestratorFunc")
+                {
+                    return;
+                }
+
+                // all overloads have the parameter orchestrator, either as an Action or a Func
+                IArgumentOperation orchestratorArgument = invocation.Arguments.First(a => a.Parameter!.Name == "orchestrator");
+                if (orchestratorArgument.Value is not IDelegateCreationOperation delegateCreationOperation)
+                {
+                    return;
+                }
+
+                IMethodSymbol? methodSymbol = null;
+                switch (delegateCreationOperation.Target)
+                {
+                    case IAnonymousFunctionOperation lambdaOperation:
+                        // use the containing symbol of the lambda (e.g. the class declaring it) as the method symbol
+                        methodSymbol = ctx.ContainingSymbol as IMethodSymbol;
+                        break;
+                    case IMethodReferenceOperation methodReferenceOperation:
+                        // use the method reference as the method symbol
+                        methodSymbol = methodReferenceOperation.Method;
+                        break;
+                    default:
+                        break;
+                }
+
+                if (methodSymbol == null)
+                {
+                    return;
+                }
+
+                // try to get the name of the orchestration from the method call, otherwise use the containing type name
+                IArgumentOperation nameArgument = invocation.Arguments.First(a => a.Parameter!.Name == "name");
+                Optional<object?> name = nameArgument.GetConstantValueFromAttribute(ctx.Operation.SemanticModel!, ctx.CancellationToken);
+                string orchestrationName = name.Value?.ToString() ?? methodSymbol.Name;
+
+                AnalyzedOrchestration orchestration = new(orchestrationName);
+
+                SyntaxNode funcRootSyntax = delegateCreationOperation.Syntax;
```

Comment:
How is `delegateCreationOperation` used? What's an example value for it?

Reply:
The framework allows orchestrators to be created using extensions methods of `DurableTaskRegistry`:
![image](https://github.com/microsoft/durabletask-dotnet/assets/13934447/f59d8d46-5de0-4f93-b8d7-8c260bcaff52)

So, after checking whether an `AddOrchestratorFunc` function was invoked, the code looks for the parameter named `orchestrator`, which as per screenshot above is either an Action or a Func, in other words, the value is an `IDelegateCreationOperation` here.

Reply:
For examples, please check those unit tests: https://github.com/allantargino/durabletask-dotnet/blob/917542335988f17a4b035b869e7d86a9708c03e4/test/Analyzers.Tests/Orchestration/DateTimeOrchestrationAnalyzerTests.cs#L225-L272